services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/GroupController.java

```diff
@@ -0,0 +1,89 @@
+package com.sourcefuse.jarc.services.usertenantservice.controller;
+
+import com.sourcefuse.jarc.services.usertenantservice.DTO.Count;
+import com.sourcefuse.jarc.services.usertenantservice.DTO.Group;
+import com.sourcefuse.jarc.services.usertenantservice.DTO.UserGroup;
+import com.sourcefuse.jarc.services.usertenantservice.auth.IAuthUserWithPermissions;
+import com.sourcefuse.jarc.services.usertenantservice.commonutils.CommonUtils;
+import com.sourcefuse.jarc.services.usertenantservice.repository.GroupRepository;
+import com.sourcefuse.jarc.services.usertenantservice.repository.UserGroupsRepository;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeanUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@RestController
+@Slf4j
+@RequestMapping("/groups")
+@RequiredArgsConstructor
+public class GroupController {
+
+    private final GroupRepository groupRepository;
+
+    private final UserGroupsRepository userGroupsRepo;
+
+    @PostMapping("")
+    public ResponseEntity<Object> createGroups(@Valid @RequestBody Group group) {
```

Comment:
Lets discuss about response entity, i dont think its necessary

Reply:
yes we can discuss